sonarqube/sonarqube-deploy.yml

```diff
@@ -0,0 +1,291 @@
+    name: "${NAME}"
+- apiVersion: v1
+  kind: RoleBinding
+  metadata:
+    name: "${NAME}_view"
+  roleRef:
+    name: view
+  subjects:
+  - kind: ServiceAccount
+    name: "${NAME}"
+- apiVersion: v1
+  kind: PersistentVolumeClaim
+  metadata:
+    name: sonarqube-data
+  spec:
+    accessModes:
+    - ReadWriteOnce
+    resources:
+      requests:
+        storage: ${SONARQUBE_PERSISTENT_VOLUME_SIZE}
+  status: {}
+- apiVersion: v1
+  kind: ImageStream
+  metadata:
+    labels:
+      name: "${NAME}"
+      app: "${NAME}"
+    name: "${NAME}"
+  spec:
+    tags:
+    - annotations:
+        openshift.io/imported-from: "${CONTAINER_IMAGE}"
+      from:
+        kind: DockerImage
+        name: "${CONTAINER_IMAGE}"
+      importPolicy: {}
+      name: latest
+      referencePolicy:
+        type: Source
+- apiVersion: v1
+  kind: DeploymentConfig
+  metadata:
+    generation: 1
+    labels:
+      app: "${NAME}"
+    name: "${NAME}"
+  spec:
+    replicas: 1
+    selector:
+      app: "${NAME}"
+      deploymentconfig: "${NAME}"
+    strategy:
+      activeDeadlineSeconds: 21600
+      recreateParams:
+        timeoutSeconds: 600
+        post:
+          execNewPod:
+            command:
+            - /bin/sh
+            - -c
+            - sleep 30 && curl http://admin:admin@${NAME}:9000/api/webhooks/create
+              -X POST -d "name=jenkins&url=${JENKINS_URL}/sonarqube-webhook/"
+            containerName: "${NAME}"
+          failurePolicy: Abort
+      type: Recreate
+    template:
+      metadata:
+        annotations:
+          openshift.io/generated-by: OpenShiftWebConsole
+        labels:
+          app: "${NAME}"
+          deploymentconfig: "${NAME}"
+      spec:
+        containers:
+        - env:
+          - name: JDBC_URL
+            value: jdbc:postgresql://sonardb:5432/sonar
+          - name: JDBC_USERNAME
+            valueFrom:
+              secretKeyRef:
+                key: database-user
+                name: sonardb
+          - name: JDBC_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                key: database-password
+                name: sonardb
+          - name: FORCE_AUTHENTICATION
+            value: "true"
+          - name: PROXY_HOST
+            value: ${PROXY_HOST}
+          - name: PROXY_PORT
+            value: ${PROXY_PORT}
+          - name: PROXY_USER
+            value: ${PROXY_USER}
+          - name: PROXY_PASSWORD
+            value: ${PROXY_PASSWORD}
+          - name: GROUP_ROLE_MAPPING
+            value: ${GROUP_ROLE_MAPPING}
+          - name: BUTTON_COLOR
+            value: ${BUTTON_COLOR}
+          imagePullPolicy: Always
+          livenessProbe:
+            failureThreshold: 3
+            httpGet:
+              path: /
+              port: 9000
+              scheme: HTTP
+            initialDelaySeconds: 45
```

Comment:
May need to increase this because the initial deployment when it creates the database schema takes longer than this and thus the container gets killed and restarted.

Reply:
On the RHPDS 3.11 cluster, setting the initial delay to 60 seconds seems to have worked.